3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -21,18 +21,84 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
+const body = document.querySelector('body');
+
+
+async function fetchData(url) {
+  try {
+    const response = await fetch(url)
+    if (!response.ok) {
+      throw new Error("SOME ERROR WITH API" +  response.status);
+    }    
+    return response.json();
+
+   } catch (error) {
+    console.log(error);
+    throw error;
+   }
+  }
+
+function fetchAndPopulatePokemons(data) {
+  //createElements
+  const select = document.createElement("select");
+  const img = document.createElement('img');
+  
+
+  // appendChild
+  body.appendChild(select);
+  body.appendChild(img);
+
+ // add options
+  const {results} = data;
+  results.map((pokemon) => {
+    const namePokemon = pokemon.name;
+    const option = document.createElement("option");
+    option.innerHTML = namePokemon;
```

Comment:
1. Try and avoid using `.innerHTML` if you just need to set the text content: use the construct that gives you just what you need and nothing more.
2. You could use the `.value` attribute to store the pokemon URL, for easy reference when the option is selected.

```suggestion
    option.textContent = pokemon.name;
    option.value = pokemon.url;
```